### PR TITLE
Tech/travis snapshot

### PR DESCRIPTION
**Pull request by brianPlummer**
- i'm not sure i'm doing the correct thing in the after_success block of the travis file.  "The goal you specified requires a project to execute but there is no POM in this directory "

### CODE REVIEW

**Comment by digitalbuddha** [comment]
@jaredsburrows mind lending a hand?

**Comment by jaredsburrows** [comment]
It looks like using `travis` command line tool changed the spacing. Why use `mvn` when this project is using `gradle`?

**Comment by digitalbuddha** [comment]
Can we just run gradlew upload archive from the after success?

**Comment by jaredsburrows** [comment]
See this blog: https://benlimmer.com/2014/01/04/automatically-publish-to-sonatype-with-gradle-and-travis-ci/. You guys have the `maven-push.gradle`. Just use `gradlew uploadArchives`. As long as there is `-SNAPSHOT` it should go to the snapshot's repo here: https://oss.sonatype.org/content/repositories/snapshots/.

**Comment by digitalbuddha** [comment]
Thank you sir!

**Comment by jaredsburrows** [comment]
Check out @JakeWharton's project here: https://github.com/JakeWharton/RxBinding/blob/master/.travis.yml#L28. He does exactly what I am talking about. `after_success` calls `gradlew uploadArchives` here https://github.com/JakeWharton/RxBinding/blob/master/.buildscript/deploy_snapshot.sh#L24.

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,5 +1,8 @@
 Change Log
 ==========
+## Version 0.1.0
+_2016-10-30_
+* introducing a no-op aar option
```
Comment (jaredsburrows) [inline CHANGELOG.md]: +1

File: travis-settings.xml [hunk 2]
```diff
@@ -0,0 +1,15 @@
+<!-- settings.xml -->
```
Comment (jaredsburrows) [inline travis-settings.xml]: Is this still necessary if you have the hidden variables in Travis?

Comment (brianPlummer) [inline travis-settings.xml]: i'll remove it!  thanks for helping us out on this! much appreciated!!

**Comment by digitalbuddha** [comment]
👍

File: travis-settings.xml (hunk 2 above)
Comment (jaredsburrows) [inline travis-settings.xml]: @brianPlummer No problem!

**Comment by jaredsburrows** [comment]
The build passes but the upload fails: https://travis-ci.org/friendlyrobotnyc/TinyDancer/builds/171853376#L1499.

**Comment by digitalbuddha** [comment]
Do you need access for snapshots. Alternatively did you add keys to proper place?

**Comment by jaredsburrows** [comment]
When making changes to the `maven-push.gradle` file, I left these variables the same: https://github.com/friendlyrobotnyc/TinyDancer/blob/master/gradle/maven-push.gradle#L19.

According to the build, https://travis-ci.org/friendlyrobotnyc/TinyDancer/builds/171854936#L147, you all are using `SONATYPE_USERNAME` and `SONATYPE_PASSWORD`.

RxBinding has a similar naming: https://github.com/JakeWharton/RxBinding/blob/master/gradle/gradle-mvn-push.gradle#L25.

I suggest jus changing the variables in `maven-push.gradle` from `NEXUS_USERNAME ` and `NEXUS_PASSWORD ` to `SONATYPE_USERNAME` and `SONATYPE_PASSWORD`, respectively.

**Comment by brianPlummer** [comment]
makes sense! trying that now!

**Comment by jaredsburrows** [comment]
Have you all tried publishing locally to test this out? Are you using the correct user/pass combo? Did you encode the correct user/pass? I am not sure why it is still failing :/.

File: gradle/maven-push.gradle [hunk 3]
```diff
@@ -16,11 +16,11 @@ def getSnapshotRepositoryUrl() {
 }
 
 def getRepositoryUsername() {
-    return hasProperty("NEXUS_USERNAME") ? NEXUS_USERNAME : ""
+    return hasProperty("SONATYPE_USERNAME") ? NEXUS_USERNAME : ""
```
Comment (jaredsburrows) [inline gradle/maven-push.gradle]: @brianPlummer Ah, this needs to be `return hasProperty("SONATYPE_USERNAME") ? SONATYPE_USERNAME : ""` and `return hasProperty("SONATYPE_PASSWORD") ? SONATYPE_PASSWORD : ""`.

Comment (brianPlummer) [inline gradle/maven-push.gradle]: changed! :)

**Comment by jaredsburrows** [comment]
Still failing: https://travis-ci.org/friendlyrobotnyc/TinyDancer#L1518.

Can you try uploading a snapshot locally?

**Comment by brianPlummer** [comment]
yeah, it's not working locally.  I'll dig into it.

**Comment by jaredsburrows** [comment]
@digitalbuddha You can just delete those `build.gradle` `versionCode` and `versionName`. See https://github.com/JakeWharton/RxBinding/blob/master/rxbinding-design/build.gradle#L27. The versions are specified here: https://github.com/friendlyrobotnyc/TinyDancer/blob/master/build.gradle#L39 which is used here: https://github.com/friendlyrobotnyc/TinyDancer/blob/master/gradle/maven-push.gradle#L34. You can test all this out using `gradlew installArchives` here: https://github.com/friendlyrobotnyc/TinyDancer/blob/master/gradle/maven-push.gradle#L109.

**Comment by digitalbuddha** [comment]
thanks for the hand holding will try again at lunch.

**Comment by jaredsburrows** [comment]
@brianPlummer @digitalbuddha Just test locally first! I think it may have to do with the travis ci setup and encrypting the username/password.

**Comment by digitalbuddha** [comment]
I'm going to merge. Im OK not having snapshot deploy work for now. We can revisit later.  Merging to get some other changes in

**Comment by jaredsburrows** [comment]
@digitalbuddha Can you at least release a new version?

**Pull request merged**